File: Count Subarrays with given XOR.cpp
class Solution {
  public:
    long subarrayXor(vector<int> &arr, int k) {
        // code here
        int n = arr.size();
        int cnt = 0;
        
        map<int, int> mp;
        int cur = 0;
        
        for(int i=0; i<n; ++i){
            
            cur ^= arr[i];
            
            if(cur==k){
                cnt++;
            }
            
            if(mp.find(cur^k)!=mp.end()){
                cnt += mp[cur^k];
            }
            mp[cur]++;
        }
        return cnt;
    }
};

File: Count Subarrays with given XOR.py
class Solution:
    def subarrayXor(self, arr, k):
        # code here
        
        prefix_xor = 0
        count = 0
        freq = {}
        
        for num in arr:
            prefix_xor ^= num
            
            if prefix_xor == k:
                count += 1
                
            recq = prefix_xor ^ k
            if recq in freq:
                count += freq[recq]
            freq[prefix_xor] = freq.get(prefix_xor,0)+ 1
        return count

